ssl/record/rec_layer_d1.c

```diff
@@ -757,9 +756,9 @@ int do_dtls1_write(SSL *s, int type, const unsigned char *buf,
      * first check if there is a SSL3_BUFFER still being written out.  This
      * will happen with non blocking IO
      */
-    if (SSL3_BUFFER_get_left(wb) != 0) {
-        OPENSSL_assert(0);      /* XDTLS: want to see if we ever get here */
-        return ssl3_write_pending(s, type, buf, len, written);
+    if (!ossl_assert(SSL3_BUFFER_get_left(wb) == 0)) {
+        SSLerr(SSL_F_DO_DTLS1_WRITE, ERR_R_INTERNAL_ERROR);
+        return 0;
```

Comment:
It seems quite strange to me that we should abort due to an internal error in a dev build, but then try and continue anyway in a non-dev build (trying to work around the fact that we should never have got here in the first place). That seems quite dangerous and the proper response is surely to fail gracefully.